### PR TITLE
[Feature-3934] Add zookeeper registry for streampark

### LINKED ISSUE #3934: [Feature] Add registy module for streampark
### Search before asking

- [X] I had searched in the [feature](https://github.com/apache/incubator-streampark/issues?q=is%3Aissue+label%3A%22Feature%22) and found no similar feature requirement.

### Description

Registrar SPI, the user can specify the registration center method and specific related parameters in the configuration file. Zookeeper is used by default

todo list:

1. Provide a unified `RegistryClient` related interface (org.apache.streampark.registry), Including registration, offline, update and other operations. The external access registration center uniformly calls the Client interface.
The registration center currently only provides two registration center(MySql & Zookeeper). We need to load the corresponding plug-in according to the user configuration. When the user does not specify it, zookeeper is used by default.

At the same time, in order to transition, we need to provide the implementation of distributed locks. (This is just for transition)

2. Implementation of Zookeeper in the registry

3. Implementation of MySql in the registry 

4. Console Server supports the registration center. The console server must share the same registry with each other, and multiple registries are not currently supported.

## Function

SP use registry to do the below three things:

1. Store the metadata of console so that it can get notify when nodes up and down.
2. Store the metadata of console to do load balance.
3. Acquire a global lock when do failover.

So for SP, the registry need to notify the server when the server subscribe data have added/deleted/updated, support a way to create/release a global lock,
delete the server's metadata when server down.

## How to use

At present, we have implements two registry: Zookeeper(Default),Mysql. If you
want to use them, you should config it at resource/application.yaml. The configuration details
can be viewed in the README of plugin under Module streampark-registry-plugins

### streampark-registry-all

This module is used for exporting the implementation of registry.
If you want to add new registry,you should add the dependency in the pom.xml

### streampark-registry-api

This module contains the relevant interfaces involved in the use of the registry.
The following are several important interfaces
1. Registry Interface: If you want to implement your own registry, you just need to implement this interface
2. ConnectionListener Interface: This interface is responsible for the connection status between the client and the registry,
The connection state can be viewed in ConnectionState.java
3. SubscribeListener Interface: This interface is responsible for monitoring the state changes of child nodes under the specified prefix.
Event content can be viewed in event.java

### streampark-registry-plugins

This module contains all registry implementations in SP

### Usage Scenario

_No response_

### Related issues

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by HxpSerein**
## What changes were proposed in this pull request

Issue Number: close https://github.com/apache/incubator-streampark/issues/3934

## Brief change log

Add zookeeper registry for streampark

## Verifying this change

This change added tests and can be verified as follows:
RegistryServiceTest

## Does this pull request potentially affect one of the following parts
 - Dependencies (does it add or upgrade a dependency): yes

### CODE REVIEW

File: streampark-console/streampark-console-service/src/main/assembly/conf/config.yaml [hunk 1]
```diff
@@ -106,6 +106,10 @@ registry:
   heartbeat-refresh-interval: 1s
   session-timeout: 3s
 
+zookeeper:
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/assembly/conf/config.yaml]: Directly configuring Zookeeper will confuse users, and they will need to learn what it does. It is recommended to change to the following:
```

high-availability:
    enable: false # true
    # The list of ZooKeeper quorum peers that coordinate the high-availability
    # setup. This must be a list of the form:
    # "host1:clientPort,host2:clientPort,..." (default clientPort: 2181)
    #
    zookeeper.quorum: 192.168.100.128:2181,192.168.100.129:2181
```

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/StreamParkConsoleBootstrap.java [hunk 2]
```diff
@@ -48,11 +52,23 @@
...
     public static void main(String[] args) throws Exception {
         new SpringApplicationBuilder()
             .properties(SpringProperties.get())
             .sources(StreamParkConsoleBootstrap.class)
             .run(args);
     }
 
+    @PostConstruct
+    public void init() {
+        registryService.start();
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/StreamParkConsoleBootstrap.java]: We need to decide whether to call this method based on the configuration value "high-availability.enable."

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/RegistryService.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+ */
+
+package org.apache.streampark.console.core.service;
+
+public interface RegistryService {
+
+    /**
+     * Start the registry service.
+     */
+    void start();
```
Comment (wolfboys) [inline streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/RegistryService.java]: It is recommended to rename the method name to `register`

File: streampark-console/streampark-console-service/src/main/assembly/conf/config.yaml (hunk 1 above)
Comment (HxpSerein) [reply streampark-console/streampark-console-service/src/main/assembly/conf/config.yaml]: changed

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/StreamParkConsoleBootstrap.java (hunk 2 above)
Comment (HxpSerein) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/StreamParkConsoleBootstrap.java]: changed

File: streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/RegistryService.java (hunk 3 above)
Comment (HxpSerein) [reply streampark-console/streampark-console-service/src/main/java/org/apache/streampark/console/core/service/RegistryService.java]: ok

**Review by wolfboys — APPROVED** [review]
LGTM

**Pull request merged**